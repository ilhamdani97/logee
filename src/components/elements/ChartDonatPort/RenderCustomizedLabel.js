import React from 'react';
import styles from './styles.scoped.css';

const RenderCustomizedLabel = ({
  cx,
  cy,
  midAngle,
  innerRadius,
  outerRadius,
  percent,
}) => {
  const RADIAN = Math.PI / 180;
  const radius = innerRadius + (outerRadius - innerRadius) * 0.5;
  const x = cx + radius * Math.cos(-midAngle * RADIAN);
  const y = cy + radius * Math.sin(-midAngle * RADIAN);

  const renderCircle = (<g className={styles.circleFill}>
    <circle
      cx={x > cx ? x+10 : x-10}
      cy={y > cy ? y+10 : y-10}
      r="15" />
    <text
      className={styles.textCircle}
      dominantBaseline="central"
      fill="black"
      textAnchor="middle"
      x={x > cx ? x+10 : x-10}
      y={y > cy ? y+10 : y-10}
    >
      {`${(percent * 100).toFixed(0)}%`}
    </text>
  </g>);

  return percent !== 0 ? renderCircle :null;
};

export default RenderCustomizedLabel;
