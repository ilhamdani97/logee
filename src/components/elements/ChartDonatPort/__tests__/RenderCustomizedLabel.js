import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import RenderCustomizedLabel from '../RenderCustomizedLabel';

describe('src/components/elements/RenderCustomizedLabel', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<RenderCustomizedLabel />);
    expect(tree).toMatchSnapshot();
  });
});
