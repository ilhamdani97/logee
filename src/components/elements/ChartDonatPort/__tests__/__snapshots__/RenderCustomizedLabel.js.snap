// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/RenderCustomizedLabel render 1`] = `
<g>
  <circle
    cx={NaN}
    cy={NaN}
    r="15"
  />
  <text
    dominantBaseline="central"
    fill="black"
    textAnchor="middle"
    x={NaN}
    y={NaN}
  >
    NaN%
  </text>
</g>
`;
